src/stateful-core.ts

```diff
@@ -24,7 +24,23 @@ import { Unsubscribe } from '@reduxjs/toolkit';
 import { isLevenshteinMatch } from './utils/searchable-facets';
 
 export default class StatefulCore {
-  constructor(private core: AnswersCore, private stateManager: StateManager) {}
+  private latestRequestIds: Record<string, number>;
+
+  constructor(private core: AnswersCore, private stateManager: StateManager) {
+    this.latestRequestIds = {};
+  }
+
+  private getNewRequestId(requestName: string): number {
+    const newId = this.latestRequestIds[requestName]
+      ? this.latestRequestIds[requestName] + 1
+      : 1;
+    this.latestRequestIds[requestName] = newId;
+    return newId;
+  }
+
+  private getLatestRequestId(requestName: string): number {
+    return this.latestRequestIds[requestName];
```

Comment:
with the current way it's used it's impossible for this to happen, but if you call getLatestRequestId for a requestName that hasn't been set yet it'll return undefined, instead of a number. I was thinking it could be better if latestRequestIds starts off initialized with
```
{
  univeralQuery: 0,
  verticalQuery: 0
//etc
}
```
and we restrict requestName to one of the possible query names, that way we don't have to worry about it and also get nicer static typing. It also lets us simplify getNewRequestId as well (no need for defaulting)